src/modules/CircularGradient/Module.js

```diff
@@ -0,0 +1,84 @@
+module.exports = function Circular(options, UI) {
+
+    var output;
+    var defaults = require('./../../util/getDefaults.js')(require('./info.json'));
+
+    // The function which is called on every draw.
+    function draw(input, callback, progressObj) {
+        progressObj.stop(true);
+        progressObj.overrideFlag = true;
+
+        var getPixels = require('get-pixels');
+        var savePixels = require('save-pixels');
+
+        var step = this;
+
+        getPixels(input.src, function(err, pixels) {
+
+            if (err) {
+                console.log("Bad Image path");
+                return;
+            }
+
+            var width = pixels.shape[0];
+            var height = pixels.shape[1];
+            console.log('achanged');
+
+            function changePixel(r, g, b, a) {
+                return [r, 0, 0, a];
+            }
+
+            function getNeighbourPixel(pixels,curX,curY,distX,distY){
+                // for (var distY = 0 ; distY < height/2 ; distY++) {
+                return [
+                    pixels.get(curX + distX, curY + distY, 0),
+                    pixels.get(curX + distX, curY + distY, 1),
+                    pixels.get(curX + distX, curY + distY, 2),
+                    pixels.get(curX + distX, curY + distY, 3),
+                ]
+                // }
+            }
+
+            var chunks = [];
+            var totalLength = 0;
+            var r = savePixels(pixels, input.format, { quality: 100 });
+
+            r.on("data", function(chunk) {
+                totalLength += chunk.length;
+                chunks.push(chunk);
+            });
+
+            r.on("end", function() {
+                var data = Buffer.concat(chunks, totalLength).toString("base64");
```

Comment:
You don't have to do the pixel saving yourself. PixelManipulation will do it for you.